test/functional/normal/K_spec.lua

```diff
@@ -31,7 +31,7 @@ describe('K', function()
 
     -- K on the text "K_spec_out" resolves to `!echo fnord >> K_spec_out`.
     feed('i'..test_file..'<ESC>K')
-    feed('<CR>') -- Press ENTER
+    retry(nil, nil, function() eq(1, eval('filereadable("'..test_file..'")')) end)
     eq({'fnord'}, eval("readfile('"..test_file.."')"))
```

Comment:
Just for reference: it's also fine to put retry() around the test assertion. It'll fail correctly/informatively.